docs/architecture/Data_caching_lineage.md

```diff
@@ -0,0 +1,297 @@
+1. Every pipeline run takes a snapshot of dataset annotations/labels, and stores it as the pipeline run's output. Such snapshot is attributed with a unique *snapshot name*.
+1. Pipeline runs must support 2 modes: with the *latest data* or in the *reproduction mode*.
+    1. When running on the *latest data*, a new snapshot of annotations/labels should be copied to the cache container, to a run-specific location defined by the *snapshot name*. The same files are uploaded to the run's output folder.
+    1. When running in the *reproduction mode*, annotations/labels are downloaded from the output storage of the run that is being reproduced. The location of the snapshot in the cache container is defined by the *snapshot name* of the run that is being reproduced.
+
+Different locations (prefixes) of the shared cache and annotation snapshots in a Premium Blob Storage container should be used for configuring different lifecycle management policies (TTL). Annotation snapshots don't need to be preserved as long as the shared cache.
+
+## Data Caching Trade Study and Recommendations
+
+Computer vision pipelines on Azure Machine Learning require transferring data from long-term storage in Azure Blob to a Premium Azure Blob container used as a cache. There are 4 major options of performing such data transfer in Azure ML pipelines:
+
+1. PythonScriptStep with custom code on top of the Azure Storage SDK or [AzCopy](https://docs.microsoft.com/en-us/azure/storage/common/storage-ref-azcopy-copy).
+1. Durable Azure Functions with custom code on top of the Azure Storage SDK or AzCopy.
+1. [DataTransferStep](https://docs.microsoft.com/en-us/python/api/azureml-pipeline-steps/azureml.pipeline.steps.data_transfer_step.datatransferstep?view=azure-ml-py), a specialized step in Azure Machine Learning, utilizes Azure Data Factory (Copy Activity).
+1. PythonScriptStep directly initiating an Azure Data Factory pipeline with [Copy Activity](https://docs.microsoft.com/en-us/azure/data-factory/copy-activity-overview).  
+
+ Option | Pros | Cons |
+| -------- | --- | ----------- |
+| PythonScriptStep with AzCopy | Fast and efficient, provides required features | Storage credentials must be exposed to the step (directly, via Key Vault, or to AML compute's identity) |<!-- markdownlint-disable MD013 -->
+| Durable Functions | Fast and efficient, provides almost all required features, no need to expose storage credentials to pipelines | Significantly increases run orchestration complexity, adds manageability overhead, no access to AML run output |<!-- markdownlint-disable MD013 -->
+| DataTransferStep | Lowest manageability overhead, storage credentials are managed within AML Datastores | Doesn't support incremental copying |
+| Azure Data Factory | no need to expose storage credentials to pipelines | Same as above + manageability overhead |<!-- markdownlint-disable MD013 -->
```

Comment:
"above" is risky in case you ever reorder/add/remove. I'd either say "same as DataTransferStep" (preferred) or copy the text.